lib/src/revset.rs

```diff
@@ -2704,6 +2704,21 @@ pub struct RevsetWorkspaceContext<'a> {
     pub workspace_id: &'a WorkspaceId,
 }
 
+/// Formats a string as symbol by quoting and escaping it if necessary
+pub fn format_symbol(literal: &str) -> String {
+    let needs_escaping = if let Ok(node) = revset_parser::parse_program(literal) {
+        !matches!(node.kind, ExpressionKind::Identifier(_))
+    } else {
+        true
+    };
+    if needs_escaping {
+        let escaped = literal.replace('\\', r#"\\"#).replace('"', r#"\""#);
```

Comment:
Maybe we can apply `char::escape_default()` to chars lower than `0x80`?
https://doc.rust-lang.org/stable/std/primitive.char.html#method.escape_default

`str::escape_debug()` might also work, but it might escape some unicode characters as `\u....` (which our parser doesn't support.)

btw, I would add `dsl_util::escape_string()` where string parser is defined.

Reply:
> > Maybe we can apply `char::escape_default()` to chars lower than `0x80`? https://doc.rust-lang.org/stable/std/primitive.char.html#method.escape_default
> 
> Why? Would it prevent some bugs? Or would it be to make the stored string not rely on unicode in order to make sure it's readable even if you have a simple terminal or font?

Just because it would be easier to implement. Suppose we want to reuse `escape_string()` for displaying purpose, I expect control characters in ASCII range are escaped.

Reply:
I assumed `char::escape_default()` would use '\xNN` for lower values, but apparently not. It is the behavior of `ascii::escape_default()`.

Reply:
I think it's nice if we can allow non-ascii values without escaping, so bookmark names like the one in #5359 are easy to read.

Reply:
> we can allow non-ascii values without escaping,

Yes, I agree. I just mean it's better to escape ASCII control characters (i.e. 0x00..=0x1f and 0x7f.)

Reply:
Oh, I see. Yes, I agree. I'll do that.

Reply:
Done now.